Source/Dafny/Rewriter.cs

```diff
@@ -118,14 +120,61 @@ public abstract class IRewriter {
       Contract.Requires(program != null);
     }
   }
-
+  /*
+   * Code Example that displays the Warnings
+   *
+   * datatype Color = Red | Green | ShadesOfGray(nat)
+      method MonochromaticMethod(c: Color) returns (x: bool) {
+      return match c
+            case Green => true
+                 ^^^ Warning: Constructor name 'Green' should contain a parenthesis at the end           
+            case anythingElse => false;
+        }
+    }
+   */
+  class ConstructorWarning : IRewriter {
+    internal override void PostResolve(Program program) {
+      base.PostResolve(program);
+      foreach (var moduleDefinition in program.Modules()) {
+        foreach (var topLevelDecl in moduleDefinition.TopLevelDecls.OfType<TopLevelDeclWithMembers>()) {
+          foreach (var callable in topLevelDecl.Members.OfType<ICallable>()) {
+            var visitor = new ConstructorWarningVisitor(this.Reporter);
+            visitor.Visit(callable, Unit.Default);
+          }
+        }
+      }
+    }
+    public ConstructorWarning(ErrorReporter reporter) : base(reporter) {
+    }
+  }
+  class ConstructorWarningVisitor : TopDownVisitor<Unit> {
+    private readonly ErrorReporter reporter;
+    public ConstructorWarningVisitor(ErrorReporter reporter) {
+      this.reporter = reporter;
+    }
+    // Implements warning for constructors in match and ensures constructor name has a set parenthesis at the end 
+    protected override bool VisitOneExpr(Expression expr, ref Unit st) {
+      if (expr is NestedMatchExpr matchExpr) {
+        var matchExprCases = matchExpr.Cases;
+        foreach (var caseExpr in matchExprCases) {
+          if (caseExpr.Pat is IdPattern idPattern) {
+            var isConstructor = idPattern.Arguments != null;
```

Comment:
How can you determine if something is a constructor there? I thought you would be looking in a dictionary of constructor name. I think the failing CI  tests come from false positives.

Reply:
This checks whether the idPattern was determined to be a constructor by the resolver, in which case `idPattern.Arguments` is given a non-null value.